roles/telemetry_verify_metrics/tasks/verify_rabbitmq_metrics.yml

```diff
@@ -1,45 +1,140 @@
-  register: result
-  delay: 30
-  retries: 10
+    # We can use "regex: namespace" to detect the old scrapeconfig. This string is absent
+    # in the new one.
+    oc get scrapeconfig {{ rabbitmq_sc_list[0].name }} -oyaml | grep 'regex: namespace'
+  register: scrapeconfig_regex_grep_result
   changed_when: false
-  until: result.rc == 0 and "rabbitmq_identity_info" in result.stdout
-  loop: "{{ cr_names.stdout_lines }}"
+  failed_when: false
+
+# TODO: Block can be deleted in the future. The fix was targeted for FR4 BR1
+- name: Perform tests without fixed rabbitmq scrapeconfigs
+  when: scrapeconfig_regex_grep_result.rc == 0
+  block:
+  - name: Verify rabbitmq scrapeconfig exists and pods are running
+    ansible.builtin.include_role:
+      name: common
+    vars:
+      common_cr_list:
+        - kind: scrapeconfigs.monitoring.rhobs
+          name: telemetry-rabbitmq
+      common_pod_list: "{{ rabbitmq_pod_list }}"
+      common_pod_status_str: "Running"
+      common_pod_nspace: openstack
+
+  - name: Check rabbitmq metric endpoints
+    ansible.builtin.shell: |
+      oc rsh openstackclient curl https://{{ item }}.openstack.svc:15691/metrics
+    register: result
+    changed_when: false
+    failed_when: result.rc >= 1
+    loop: "{{ cr_names.stdout_lines }}"
+
+  - name: |
+      TEST Use openstack observabilityclient to verify rabbitmq metrics are stored in prometheus
+    ansible.builtin.shell: |
+      {{ openstack_cmd }} metric query rabbitmq_identity_info{instance=\'{{ item }}.openstack.svc:15691\'} --disable-rbac
+    register: result
+    delay: 30
+    retries: 10
+    changed_when: false
+    until: result.rc == 0 and "rabbitmq_identity_info" in result.stdout
+    loop: "{{ cr_names.stdout_lines }}"
+
+- name: Perform tests with fixed rabbitmq scrapeconfigs
+  when: scrapeconfig_regex_grep_result.rc >= 1
+  block:
+  - name: Verify rabbitmq scrapeconfigs exist and pods are running
+    ansible.builtin.include_role:
+      name: common
+    vars:
+      common_cr_list: "{{ rabbitmq_sc_list }}"
+      common_pod_list: "{{ rabbitmq_pod_list }}"
+      common_pod_status_str: "Running"
+      common_pod_nspace: openstack
+
+  - name: Retrieve rabbitmq pod IPs
+    ansible.builtin.shell: |
+      oc get pods {{ item }} -ojsonpath --template \{.status.podIP\}
+    register: result
+    changed_when: false
+    failed_when: result.rc >= 1
+    loop: "{{ rabbitmq_pod_list }}"
+
+  - name: Create a list of rabbitmq pod IPs
+    ansible.builtin.set_fact:
+      rabbitmq_pod_ips: "{{ rabbitmq_pod_ips | default([]) + [item.stdout] }}"
+    changed_when: false
+    loop: "{{ result.results }}"
+
+  - name: Check rabbitmq metric endpoints
+    ansible.builtin.shell: |
+      oc rsh openstackclient curl https://{{ item }}:15691/metrics --insecure
+    register: result
+    changed_when: false
+    failed_when: result.rc >= 1
+    loop: "{{ rabbitmq_pod_ips }}"
```

Comment:
Could you loop through results.results here instead of having a separate set_fact?